src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -616,7 +608,15 @@ func (a *StorageMinerActorCode_I) ProveCommitSector(rt Runtime, info sector.Sect
 		// expired
 		delete(st.PreCommittedSectors(), preCommitSector.Info().SectorNumber())
 		UpdateRelease(rt, h, st)
-		return rt.ErrorReturn(exitcode.UserDefinedError(0)) // TODO: user dfined error code?
+
+		TODO()
+		// TODO: if we abort, the above state changes (delete...) will not be committed.
+		// Is this desired? (alternative: support method failure while still committing
+		// state changes if UpdateRelease has already been called)
```

Comment:
@zixuanzh can you comment on the necessity or otherwise of this cleanup code? Can/will it be cleaned up somewhere else if not here? I think it should be the same whether a miner submits a too-late proof or withholds it indefinitely.

Reply:
ProveCommit should fail when it arrives later than MAX_PROVE_COMMIT. I agree that garbage collection (together with some slashing for missing MAX_PROVE_COMMIT) should happen whether ProveCommit arrives too late or it never happens but I am not sure where and how we should do this. We also need to clean up deals in these PreCommitted sectors.

Reply:
Presumably some amount of cleanup will need to happen via cron anyway, if miners withhold proofs indefinitely, so maybe we can add a TODO in cron to implement this (and any other) cleanup there, and remove the cleanup here?

Reply:
(Just updated PR with this change.)